compiler/ballerina-lang/src/main/java/io/ballerina/projects/Project.java

```diff
@@ -97,5 +100,52 @@ protected void populateCompilerContext() {
 
     public abstract DocumentId documentId(Path file);
 
+    public abstract Optional<Path> documentPath(DocumentId documentId);
+
+    /**
+     * Returns the path of the specified file from dependency bala.
+     *
+     * @param org package org of the dependency
+     * @param moduleName module name of the dependency
+     * @param filename name of the file to get the path of
+     * @return path of the file
+     * @throws ProjectException if file cannot be found in the dependency package
+     */
+    public Path dependencyFilePath(String org, String moduleName, String filename) {
```

Comment:
As per chat with @azinneera, We decided to remove this method from the `Project` class as this is not the correct place.

Reply:
Removed this method in 5655dd8